src/routers/__tests__/StackRouter-test.js

```diff
@@ -856,6 +856,139 @@ describe('StackRouter', () => {
     ]);
   });
 
+  test('Handles the navigate action with params and nested StackRouter as a first action', () => {
+    const state = TestStackRouter.getStateForAction({
+      type: NavigationActions.NAVIGATE,
+      routeName: 'main',
+      params: {
+        code: 'test',
+        foo: 'bar',
+      },
+      action: {
+        type: NavigationActions.NAVIGATE,
+        routeName: 'profile',
+        params: {
+          id: '4',
+          code: 'test',
+          foo: 'bar',
+        },
+        action: {
+          type: NavigationActions.NAVIGATE,
+          routeName: 'list',
+          params: {
+            id: '10259959195',
+            code: 'test',
+            foo: 'bar',
+          },
+        },
+      },
+    });
+
+    expect(state && state.index).toEqual(0);
+    expect(state && state.isTransitioning).toEqual(false);
+    /* $FlowFixMe */
```

Comment:
flow is gone, please rm!

Reply:
Done. Also added #3439 to remove the reset of these in the codebase.